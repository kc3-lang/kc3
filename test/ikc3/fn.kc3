quote fn (x) { x }
quote fn (x, _y) { x }
quote fn ([x | _y]) { x }
quote fn {
  ([]) { :error }
  ([x | _y]) { x }
  (_) { :error }
}
quote fn (x) do
  "Hello, world !"
  x * 2
end
a = fn (x) { x }
a(1)
b = fn (x, _y) { x }
b(1, 2)
c = fn ([x | _y]) { x }
c([1, 2])
d = fn {
  ([]) { :error }
  ([x | _y]) { x }
  (_) { :error2 }
}
d([1, 2])
e = fn (x) do
  "Hello, world !"
  x * 2
end
e(2)
quote do
  f = fn (x) do
    puts("Hello, world !")
    x * 2
  end
  f(2)
end
do
  f = fn (x) do
    puts("Hello, world !")
    x * 2
  end
  f(2)
end
quote f = fn (x) { fn (y) { x * y } }
f = fn (x) { fn (y) { x * y } }
quote g = f(2)
g = f(2)
quote g(3)
g(3)
quote h = fn (x) {
  unwind_protect(do
    puts("a")
    return x * 2
    -1
  end, puts("b"))
}
h = fn (x) {
  unwind_protect(do
    puts("a")
    return x * 2
    -1
  end, puts("b"))
}
quote h(21)
h(21)
