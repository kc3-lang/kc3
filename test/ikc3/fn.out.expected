fn (x) { x }
fn (x, _y) { x }
fn ([x | _y]) { x }
fn {
  ([]) { :error }
  ([x | _y]) { x }
  (_) { :error }
}
fn (x) do
  "Hello, world !"
  x * 2
end
fn (x) { x }
1
fn (x, _y) { x }
1
fn ([x | _y]) { x }
1
fn {
  ([]) { :error }
  ([x | _y]) { x }
  (_) { :error2 }
}
1
fn (x) do
  "Hello, world !"
  x * 2
end
4
